src/puppetlabs/puppetdb/query_eng/engine.clj

```diff
@@ -1087,10 +1112,25 @@
             (map->InExpression {:column (columns->fields query-rec (maybe-vectorize column))
                                 :subquery (user-node->plan-node query-rec subquery-expression)})
 
+            [["extract" [["function" & fargs]] expr]]
+            (let [[f & args] fargs
+                  args' (if args args :*)]
```

Comment:
Also I might this about moving the default behavior to the honeysql-from-query function so you can dedup the logic here.